### PR TITLE
feat: better default json schemas

### LINKED ISSUE #1673: [Feature]: include common json schema
### Feature motivation

we should have them

### Describe the expected solution

add this to the json lsp setup section:

```
-- https://www.schemastore.org/json/
local M = {}

M.setup = function()
  local schemas = {
    {
      description = "Package JSON file",
      fileMatch = { "package.json" },
      url = "https://json.schemastore.org/package.json",
    },
    {
      description = "TypeScript compiler configuration file",
      fileMatch = { "tsconfig.json", "tsconfig.*.json" },
      url = "http://json.schemastore.org/tsconfig",
    },
    {
      description = "Lerna config",
      fileMatch = { "lerna.json" },
      url = "http://json.schemastore.org/lerna",
    },
    {
      description = "Babel configuration",
      fileMatch = { ".babelrc.json", ".babelrc", "babel.config.json" },
      url = "http://json.schemastore.org/lerna",
    },
    {
      description = "ESLint config",
      fileMatch = { ".eslintrc.json", ".eslintrc" },
      url = "http://json.schemastore.org/eslintrc",
    },
    {
      description = "Bucklescript config",
      fileMatch = { "bsconfig.json" },
      url = "https://bucklescript.github.io/bucklescript/docson/build-schema.json",
    },
    {
      description = "Prettier config",
      fileMatch = { ".prettierrc", ".prettierrc.json", "prettier.config.json" },
      url = "http://json.schemastore.org/prettierrc",
    },
    {
      description = "Vercel Now config",
      fileMatch = { "now.json" },
      url = "http://json.schemastore.org/now",
    },
    {
      description = "Stylelint config",
      fileMatch = { ".stylelintrc", ".stylelintrc.json", "stylelint.config.json" },
      url = "http://json.schemastore.org/stylelintrc",
    },
  }

  local function extend(tab1, tab2)
    for _, value in ipairs(tab2) do
      table.insert(tab1, value)
    end
    return tab1
  end

  local extended_schemas = extend(schemas, require("nlspsettings.jsonls").get_default_schemas())

  lvim.lang.json.lsp.setup.settings = {
    json = {
      schemas = extended_schemas,
    },
  }
end

return M
```

### Describe the alternatives you've considered

none

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by abzcoding**
# Description

better json schema defaults

Fixes #1673

### CODE REVIEW

**Comment by ChristianChiarulli** [comment]
only thing to test is to make sure it still works with nlspconfig

**Comment by abzcoding** [comment]
Screen Shot 2021-10-04 at 11 32 00 AM
>

**Comment by kylo252** [comment]
~~Can you try to see if we can skip all the ones from the json-store and let the server automatically download it instead?~~

Nope, not yet anyway: https://github.com/microsoft/vscode/issues/26289

File: lua/lsp/providers/jsonls.lua [hunk 1]
```diff
@@ -1,20 +1,200 @@
...
-  schemas = jsonls_settings.get_default_schemas()
+  default_schemas = jsonls_settings.get_default_schemas()
 end
 
+local schemas = {
+  {
+    description = "Package JSON file",
+    fileMatch = { "package.json" },
+    url = "https://json.schemastore.org/package.json",
+  },
```
Comment (kylo252) [inline lua/lsp/providers/jsonls.lua:12]: Duplicate entry.

File: lua/lsp/providers/jsonls.lua [hunk 2]
```diff
@@ -1,20 +1,200 @@
-local schemas = nil
+local default_schemas = nil
 local status_ok, jsonls_settings = pcall(require, "nlspsettings.jsonls")
 if status_ok then
-  schemas = jsonls_settings.get_default_schemas()
+  default_schemas = jsonls_settings.get_default_schemas()
 end
 
+local schemas = {
+  {
```
Comment (kylo252) [inline lua/lsp/providers/jsonls.lua:8]: Duplicate entry.

File: lua/lsp/providers/jsonls.lua [hunk 3]
```diff
@@ -1,20 +1,200 @@
...
+      ".eslintrc.json",
+      ".eslintrc",
+    },
+    url = "http://json.schemastore.org/eslintrc",
+  },
+  {
+    description = "Bucklescript config",
+    fileMatch = { "bsconfig.json" },
+    url = "https://bucklescript.github.io/bucklescript/docson/build-schema.json",
+  },
```
Comment (kylo252) [inline lua/lsp/providers/jsonls.lua:47]: 404. What's even Bucklescript 😅 

from: http://schemastore.org/api/json/catalog.json

```json
    {
      "name": "bucklescript",
      "description": "BuckleScript configuration file",
      "fileMatch": [
        "bsconfig.json"
      ],
      "url": "https://raw.githubusercontent.com/rescript-lang/rescript-compiler/8.2.0/docs/docson/build-schema.json"
    },
```

**Review by kylo252 — CHANGES_REQUESTED** [review]
The only real note is a quick find-and-replace `http://` -> `https://` for the Github and SchemaStore links.

**Comment by abzcoding** [comment]
@kylo252 just updated the code based on your review

**Review by kylo252 — APPROVED** [review]

File: lua/lsp/providers/jsonls.lua [hunk 4]
```diff
@@ -1,20 +1,195 @@
...
+  end
+  return tab1
+end
+
+local extended_schemas = extend(schemas, default_schemas)
+
 local opts = {
   setup = {
     settings = {
       json = {
```
Comment (PasiBergman) [inline lua/lsp/providers/jsonls.lua:191]: `settings` should be moved outside of `setup` =>
```lua
local opts = {
  settings = {
    json = {
      schemas = schemas,
    },
  },
  setup = {
    commands = {
       ....
    },
  },
}
```

Comment (PasiBergman) [reply lua/lsp/providers/jsonls.lua:191]: Sorry. Didn't notice PR was merged.

Comment (abzcoding) [reply lua/lsp/providers/jsonls.lua:191]: no worries mate, i did fix it in https://github.com/LunarVim/LunarVim/pull/1682

**Pull request merged**